llvm/test/Transforms/SimplifyCFG/switch-to-select.ll

```diff
@@ -0,0 +1,341 @@
+  ret float %retVal
+}
+
+define float @two_instructions_in_bb(i32 %a, i32 %b, float %c, float %d) {
+; ALL-LABEL: @two_instructions_in_bb(
+; ALL-NEXT:  .entry:
+; ALL-NEXT:    [[TMP0:%.*]] = icmp eq i32 [[A:%.*]], 1
+; ALL-NEXT:    br i1 [[TMP0]], label [[SW:%.*]], label [[RETURN:%.*]]
+; ALL:       sw:
+; ALL-NEXT:    [[SCALE_I0:%.*]] = fmul reassoc nnan nsz arcp contract afn float [[C:%.*]], 2.000000e+00
+; ALL-NEXT:    [[DOTI0:%.*]] = fadd reassoc nnan nsz arcp contract afn float [[SCALE_I0]], -1.000000e+00
+; ALL-NEXT:    [[DOTI1:%.*]] = fadd reassoc nnan nsz arcp contract afn float [[D:%.*]], -3.000000e+00
+; ALL-NEXT:    switch i32 [[B:%.*]], label [[RETURN]] [
+; ALL-NEXT:      i32 0, label [[SW_BB1:%.*]]
+; ALL-NEXT:      i32 1, label [[SW_BB2:%.*]]
+; ALL-NEXT:      i32 2, label [[SW_BB3:%.*]]
+; ALL-NEXT:      i32 3, label [[SW_BB4:%.*]]
+; ALL-NEXT:    ]
+; ALL:       sw.bb1:
+; ALL-NEXT:    br label [[RETURN]]
+; ALL:       sw.bb2:
+; ALL-NEXT:    br label [[RETURN]]
+; ALL:       sw.bb3:
+; ALL-NEXT:    [[TMP1:%.*]] = fneg reassoc nnan nsz arcp contract afn float [[DOTI0]]
+; ALL-NEXT:    br label [[RETURN]]
+; ALL:       sw.bb4:
+; ALL-NEXT:    [[TMP2:%.*]] = fneg reassoc nnan nsz arcp contract afn float [[DOTI0]]
+; ALL-NEXT:    [[TMP3:%.*]] = fneg reassoc nnan nsz arcp contract afn float [[DOTI1]]
+; ALL-NEXT:    br label [[RETURN]]
+; ALL:       return:
+; ALL-NEXT:    [[SAMPLEPOS_1_I0:%.*]] = phi float [ 0.000000e+00, [[DOTENTRY:%.*]] ], [ [[TMP2]], [[SW_BB4]] ], [ -1.000000e+00, [[SW_BB3]] ], [ [[DOTI0]], [[SW_BB2]] ], [ 1.000000e+00, [[SW_BB1]] ], [ 4.000000e+00, [[SW]] ]
+; ALL-NEXT:    ret float [[SAMPLEPOS_1_I0]]
+;
+.entry:
+  %5 = icmp eq i32 %a, 1
+  br i1 %5, label %sw, label %return
+
+sw:
+  %scale.i0 = fmul reassoc nnan nsz arcp contract afn float %c, 2.000000e+00
+  %.i0 = fadd reassoc nnan nsz arcp contract afn float %scale.i0, -1.000000e+00
+  %.i1 = fadd reassoc nnan nsz arcp contract afn float %d, -3.000000e+00
+  switch i32 %b, label %return [
+  i32 0, label %sw.bb1
+  i32 1, label %sw.bb2
+  i32 2, label %sw.bb3
+  i32 3, label %sw.bb4
+  ]
+sw.bb1:                                               ; preds = %sw
+  br label %return
+
+sw.bb2:                                               ; preds = %sw
+  br label %return
+
+sw.bb3:                                               ; preds = %sw
+  %18 = fneg reassoc nnan nsz arcp contract afn float %.i0
+  br label %return
+
+sw.bb4:                                               ; preds = %sw
+  %20 = fneg reassoc nnan nsz arcp contract afn float %.i0
+  %21 = fneg reassoc nnan nsz arcp contract afn float %.i1
+  br label %return
+return:                                               ; preds = %sw.bb1, %sw.bb2, %sw.bb3, %sw.bb4, %sw, %.entry
+  %samplePos.1.i0 = phi float [ 0.000000e+00, %.entry ], [ %20, %sw.bb4 ], [ -1.000000e+00, %sw.bb3 ], [ %.i0, %sw.bb2 ], [ 1.000000e+00, %sw.bb1 ], [ 4.000000e+00, %sw ]
+  ret float %samplePos.1.i0
+}
```

Comment:
Negative tests? For example,
- No common dest for cases.
- The switch jumps direct into the dest block.
- Loop with switch.

Reply:
Can you please add these tests?